### PR TITLE
Loosen type condition in basic_math

### PR DESCRIPTION
**Pull request by okuta**
The basic math functions accept an array of `numpy.float16`, `numpy.float32` and `numpy.float64`.
See also  #555.

### CODE REVIEW

File: tests/chainer_tests/functions_tests/math_tests/test_basic_math.py [hunk 1]
```diff
@@ -816,19 +818,19 @@ def test_mul_backward_cpu(self):
     def test_rmul_backward_cpu(self):
         self.backward_cpu(lambda x, y: y * x)
 
-    @condition.retry(3)
+    @condition.retry(10)
```
Comment (delta2323) [inline tests/chainer_tests/functions_tests/math_tests/test_basic_math.py]: Do you have any reason why you change the number of retry?

File: chainer/functions/math/basic_math.py [hunk 2]
```diff
@@ -192,8 +192,7 @@ def label(self):
     def check_type_forward(self, in_types):
         type_check.expect(in_types.size() == 2)
         type_check.expect(
-            in_types[0].dtype == numpy.float32,
-            in_types[1].dtype == numpy.float32,
+            in_types[0].dtype == in_types[1].dtype,
```
Comment (delta2323) [inline chainer/functions/math/basic_math.py]: Is it intentional that you do not impose `in_types[0].dtype.kind == 'f'`, which is added in `Div`, `PowVarVar` etc.

File: tests/chainer_tests/functions_tests/math_tests/test_basic_math.py [hunk 3]
```diff
@@ -176,55 +184,40 @@ def test_mul_backward_gpu(self):
...
-    def make_data(self):
-        x1 = numpy.random.uniform(.5, 1, ()).astype(numpy.float32)
-        x2 = numpy.random.uniform(.5, 1, ()).astype(numpy.float32)
-        gy = numpy.random.uniform(-1, 1, ()).astype(numpy.float32)
-        return x1, x2, gy
-
-
+@testing.parameterize(*testing.product({
+    'dtype': [numpy.float16, numpy.float32, numpy.float64],
+}))
```
Comment (delta2323) [inline tests/chainer_tests/functions_tests/math_tests/test_basic_math.py]: Please add parameterized test with respect to `shape: [(3, 2), ()]`

File: tests/chainer_tests/functions_tests/math_tests/test_basic_math.py [hunk 4]
```diff
@@ -582,12 +593,12 @@ def test_div_backward_cpu(self):
...
 
-    @condition.retry(3)
+    @condition.retry(10)
     def test_pow_backward_cpu(self):
         self.backward_cpu(lambda x, y: x ** y)
 
-    @condition.retry(3)
-    def test_rpow_backward_cpu(self):
+    @condition.retry(10)
+    def s(self):
```
Comment (delta2323) [inline tests/chainer_tests/functions_tests/math_tests/test_basic_math.py]: Please restore the original function name.

File: tests/chainer_tests/functions_tests/math_tests/test_basic_math.py (hunk 1 above)
Comment (okuta) [inline tests/chainer_tests/functions_tests/math_tests/test_basic_math.py]: For stabilization of test

**Comment by delta2323** [comment]
LGTM. We will merge this PR at next minor release.

**Pull request merged**